README.md

```diff
@@ -50,59 +43,54 @@ public class SampleApplication {
 
 For a complete example, see `mesos-starter-sample` module.
 
-## Starting the application
-The only required parameter for the scheduler is `mesos.master`. The value of the parameter is passed directly to the Mesos Scheduler Driver which allows the following formats
-
-- `host:port`
-- `zk://host1:port1,host2:port2,.../path`
-- `zk://username:password@host1:port1,host2:port2,.../path`
-- `file:///path/to/file`
-
-For resiliency it is also recommended to point the application to a Zookeeper cluster and giving the instance a name.
-```
-mesos.zookeeper.server=zookeeper:2181
-mesos.framework.name=sampleApp1
-```
-
-The purpose of `mesos.framework.name` is to distinguish instances of the scheduler.
+# Configuration
+Most of an application's boilerplate is provided my Mesos-Starter. To configure that boilerplate, there are a range of configuration options. This table describes those options, whether they are optional and whether they have a default value.
+
+Command                         | Description                                           | Default               | Required
+---                             | ---                                                   | ---                   | ---
+`spring.application.name`       | Application name                                      |                       | Yes
+`mesos.framework.name`          | Framework name used in Mesos and ZooKeeper            |                       | Yes
+`mesos.master`                  | Path to mesos master                                  |                       | Yes
+`mesos.zookeeper.server`        | IP:PORT of the zookeeper server                       |                       | Yes
+`mesos.resources.cpus`          | CPUs allocated to the task                            | Accept all            |
+`mesos.resources.mem`           | RAM allocated to the task                             | Accept all            |
+`mesos.resources.count`         | Number of task instances                              | As many as possible   |
+`mesos.resources.ports`         | A map of port mappings                                |                       |
+`mesos.resources.distinctSlave` | Reject offers from hosts that already have a task     | `false`               |
```

Comment:
Looking at it. I think we should change default to `true` so that if you're missing `mesos.resources.count` it won't create a million tasks.